File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    def dfs(curr, idx):
        if idx == len(numbers):
            if target == curr:
                return 1
            return 0
        
        return dfs(curr+numbers[idx], idx+1) + dfs(curr-numbers[idx], idx+1)
    
    return dfs(0, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    private int[] numbers;
    private int target;
    
    public int solution(int[] numbers, int target) {
        this.numbers = numbers;
        this.target = target;        
        
        return this.dfs(0, 0);
    }
    
    public int dfs(int curr, int idx) {
        
        if (idx == numbers.length) {
            if (target == curr) return 1;
            else return 0;
        }
        
        return dfs(curr + numbers[idx], idx + 1) + dfs(curr - numbers[idx], idx + 1);
    }
}